tests/run_spark_hdfs.py

```diff
@@ -193,31 +193,83 @@ def cleanup_and_exit(namenode, tmpdir, exit_code):
             pid = subprocess.Popen('hadoop fs -rm -r '+namenode+'/home/hadoop/.tiledb/', shell=True, stdout=subprocess.PIPE);
     sys.exit(exit_code);
 
+def print_error_and_exit(namenode, tmpdir, stdout_string, stderr_string):
+    sys.stderr.write('Stdout: '+stdout_string+'\n')
+    sys.stderr.write('Stderr: '+stderr_string+'\n')
+    cleanup_and_exit(namenode, tmpdir, -1)
+
+def substitute_workspace_dir(jsonfile, wsdir):
+    import fileinput
+    for line in fileinput.input(jsonfile, inplace=True):
+        print(line.replace("#WORKSPACE_DIR#", wsdir))
+
+def sanity_test_spark_bindings(tmpdir, lib_path, jar_dir, jacoco, genomicsdb_version, spark_master, spark_deploy, namenode):
+    sys.stdout.write("Sanity testing Spark Bindings...")
+    import tarfile
+    sanity_test_dir = tmpdir+'/sanity_test'
+    tar = tarfile.open('inputs/sanity.test.gz', 'r:gz')
+    tar.extractall(path=sanity_test_dir)
+    loader_json = sanity_test_dir+'/t0_1_2.json'
+    query_json = sanity_test_dir+'/query.json'
+    querypb_json = sanity_test_dir+'/querypb.json'
+    substitute_workspace_dir(loader_json, sanity_test_dir)
+    substitute_workspace_dir(query_json, sanity_test_dir)
+    substitute_workspace_dir(querypb_json, sanity_test_dir)
+
+    # Expected exception when run without json files
+    spark_cmd = 'spark-submit --master '+spark_master+' --deploy-mode '+spark_deploy+' --total-executor-cores 1 --executor-memory 512M  --conf "spark.executor.extraJavaOptions='+jacoco+'" --conf "spark.driver.extraJavaOptions='+jacoco+'" --class org.genomicsdb.spark.api.GenomicsDBSparkBindings '+jar_dir+'/genomicsdb-'+genomicsdb_version+'-allinone.jar'
+    try:
+        pid = subprocess.Popen(spark_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout_string, stderr_string = pid.communicate()
+        if(pid.returncode == 0):
+            sys.stderr.write('Spark Query : '+spark_cmd+'. Expected a failure, but the test succeeded\n')
+            print_error_and_exit(namenode, tmpdir, stdout_string, stderr_string)
+    except:
```

Comment:
Does Popen pass along the exception from the spark_cmd somehow? If so, can we check for the particular exception that we'll get here?

My guess is that pid.returncode will be non-zero, so you probably want move line 228 to an else condition starting after 226 and get rid of the try/except.

Reply:
Removed the try/except.